src/backend/distributed/executor/multi_utility.c

```diff
@@ -835,6 +863,98 @@ ErrorIfDistributedRenameStmt(RenameStmt *renameStatement)
 
 
 /*
+ * CreateLocalTable gets DDL commands from the remote node for the given
+ * relation. Then, it creates the local relation as temporary and on commit drop.
+ */
+static void
+CreateLocalTable(RangeVar *relation, char *nodeName, int32 nodePort)
+{
+	List *ddlCommandList = NIL;
+	ListCell *ddlCommandCell = NULL;
+
+	char *relationName = relation->relname;
+	char *schemaName = relation->schemaname;
+	char *qualifiedName = quote_qualified_identifier(schemaName, relationName);
+
+	/* fetch the ddl commands needed to create the table */
+	StringInfo tableNameStringInfo = makeStringInfo();
+	appendStringInfoString(tableNameStringInfo, qualifiedName);
+
+	/*
+	 * The warning message created in TableDDLCommandList() is descriptive
+	 * enough; therefore, we just throw an error which says that we could not
+	 * run the copy operation.
+	 */
+	ddlCommandList = TableDDLCommandList(nodeName, nodePort, tableNameStringInfo);
+	if (ddlCommandList == NIL)
+	{
+		ereport(ERROR, (errmsg("could not run copy from the worker node")));
+	}
+
+	/* apply DDL commands against the local database */
+	foreach(ddlCommandCell, ddlCommandList)
+	{
+		StringInfo ddlCommand = (StringInfo) lfirst(ddlCommandCell);
+		Node *ddlCommandNode = ParseTreeNode(ddlCommand->data);
+		CreateStmt *createStatement = NULL;
+
+		if (IsA(ddlCommandNode, CreateStmt) ||
+			IsA(ddlCommandNode, CreateForeignTableStmt))
+		{
+			if (IsA(ddlCommandNode, CreateStmt))
+			{
+				createStatement = (CreateStmt *) ddlCommandNode;
+			}
+			else
+			{
+				CreateForeignTableStmt *createForeignTableStmt =
+					(CreateForeignTableStmt *) ddlCommandNode;
+				createStatement = (&createForeignTableStmt->base);
+			}
+
+			/* create the local relation as temporary and on commit drop */
+			createStatement->relation->relpersistence = RELPERSISTENCE_TEMP;
+			createStatement->oncommit = ONCOMMIT_DROP;
+
+			/* temporarily strip schema name */
+			createStatement->relation->schemaname = NULL;
+		}
+		else if (IsA(ddlCommandNode, CreateForeignServerStmt))
+		{
+			CreateForeignServerStmt *createServerStmt =
+				(CreateForeignServerStmt *) ddlCommandNode;
+			if (GetForeignServerByName(createServerStmt->servername, true) != NULL)
+			{
+				/* server already exists, silently skip */
+				continue;
+			}
+		}
+		else if (IsA(ddlCommandNode, IndexStmt) ||
```

Comment:
Maybe just a generic else {? If we ever add a DDL command (like create trigger), we're more likely to not want it here.